src/memory/HeapAllocator.cpp

```diff
@@ -324,7 +324,7 @@ void HeapAllocator::reserve(size_t sizeToAllocate) {
     void* memory = mBaseAllocator.allocate(sizeToAllocate + sizeof(MemoryUnitHeader));
     assert(memory != nullptr);
 
-    // Check that allocated memory is 16-bytes aligned
+    // Check that allocated memory is 8-bytes aligned
     assert(reinterpret_cast<uintptr_t>(memory) % GLOBAL_ALIGNMENT == 0);
```

Comment:
This was the assertion that was failing. The memory address was aligned to 8 not 16 (when GLOBAL_ALIGNMENT was 16)

Reply:
Can you please provide a document/link or something saying that 8 bytes memory alignment is required on Android? I want to make sure this is not a bug or something else that is wrong. I don't want to change this value just because it solves an issue on your side.

Reply:
I can't find any documents or links that says 8 bytes memory alignment is required on Android. However I just know from experience getting my Android NDK app up and running that 8 bytes is optimal.
The assertion that was failing was `HeapAllocator.cpp:328`

Reply:
If this assert was failing when GLOBAL_ALIGMNENT =16, this could be a bug. It means that the previous call:

https://github.com/DanielChappuis/reactphysics3d/blob/3da2a2cba0cf9b783b80a041a7c99d0b4c2e468d/src/memory/HeapAllocator.cpp#L324

failed to allocated 16 bytes aligned memory. That's strange because this statement should call the following line:

https://github.com/DanielChappuis/reactphysics3d/blob/3da2a2cba0cf9b783b80a041a7c99d0b4c2e468d/include/reactphysics3d/memory/DefaultAllocator.h#L67

Can you check if this line with the call to std::aligned_alloc() really returns memory that is not 16 bytes aligned when GLOBAL_ALIGNMENT=16? This mean that there is something wrong with this standard library method on Android or maybe there is another method that should be used on Android to get aligned memory but the solution is probably not to change GLOBAL_ALIGNMENT to 8 bytes.

Reply:
I'll have another debugging session later on today and get back to you :)

Reply:
Hello. I know when spoke about this last year but do you have any news regarding this issue?

Reply:
I don't sadly. I have gone through various Physics Engines in the last year, currently settling on Bullet.

Reply:
Ok thanks for your feedback.